src/components/functions/forms/GenericForm.js

```diff
@@ -331,15 +311,15 @@ class FormGenerator extends React.Component {
             const fValue = parseFloat(value);
 
             if (!regObj.test(value)) {
-                callback("Bad value!");
+                callback("Incorrect format!");
             }
 
             if (validationRules.min && fValue < parseFloat(validationRules.min)) {
-                callback("Bad value!");
+                callback("Minimum value is: " + validationRules.min);
             }
 
             if (validationRules.max && fValue > parseFloat(validationRules.max)) {
-                callback("Bad value!");
+                callback("Maximum value is: " + validationRules.max);
```

Comment:
I have only changed all messages for single inputs, not for `entry_module`. It will be done in #90 because it requires more work.

About this message `Incorrect format` - I used regular expressions to determine if this is float or int value. I am not handing negative values, so "-2" failed on pattern matching for float regexp. Should we allow negative numbers?

Reply:
Yes, negative values should be allowed. There are some physical quantities where both start and end can be negative. You can also provide support for negative values in separate issue. Handling scientific format (i.e. 1e9) could be also nice.